src/Entity/CitationType.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Drupal\stanford_publication\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBundleBase;
+
+/**
+ * Defines the Citation type entity.
+ *
+ * @ConfigEntityType(
+ *   id = "citation_type",
+ *   label = @Translation("Citation type"),
+ *   handlers = {
+ *     "view_builder" = "Drupal\Core\Entity\EntityViewBuilder",
+ *     "list_builder" = "Drupal\stanford_publication\CitationTypeListBuilder",
+ *     "form" = {
+ *       "add" = "Drupal\stanford_publication\Form\CitationTypeForm",
+ *       "edit" = "Drupal\stanford_publication\Form\CitationTypeForm",
+ *       "delete" = "Drupal\stanford_publication\Form\CitationTypeDeleteForm"
+ *     },
+ *     "route_provider" = {
+ *       "html" = "Drupal\Core\Entity\Routing\AdminHtmlRouteProvider",
+ *     },
+ *   },
+ *   config_prefix = "citation_type",
+ *   admin_permission = "administer site configuration",
+ *   bundle_of = "citation",
+ *   entity_keys = {
+ *     "id" = "id",
```

Comment:
Revisions?

Reply:
Not on citation types. that's the bundle